src/main/java/aztech/modern_industrialization/util/TransferHelper.java

```diff
@@ -77,19 +78,32 @@ public static ItemStack extractMatching(IItemHandler src, Predicate<ItemStack> p
             var stack = src.getStackInSlot(slot);
             if (predicate.test(stack)) {
                 ret = src.extractItem(slot, maxAmount, false);
+            } else if (recursive) {
+                var capability = stack.getCapability(Capabilities.ItemHandler.ITEM);
```

Comment:
I don't think that's correct: the stack coming from `getStackInSlot` should not be modified in any way. Might be fine for player inventories, but then it shouldn't be in a general purpose method like this.

I think that trying to extract from the item inside the item handler is too complex, so maybe this should be rewritten somehow to operate at the `Inventory` level?